mautic/mautic
# [Enhancement] Add field type for money, and migrate Attribution to it.

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? | Yes
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | fixes #5684
| BC breaks? | ?
| Deprecations? | 

[//]: # ( Required: )
#### Description:
I just wanted "Attribution" to be stored as a DECIMAL(19, 4) instead of DOUBLE so that we don't end up with weird rounding errors of thousandths of cents that make our finance people cringe... but then went down the rabbit hole and realized we really need a "Money" field type for that. So this PR adds a field type for "Money" and migrates the Attribution field over.

![Money field creation screen](https://i.imgur.com/JcxbE0L.png)

#### Steps to test this PR:
1. Run unit tests.
2. Create a custom field of type "Money", and ensure the field functions.
3. Check for rounding errors.

#### List backwards compatibility breaks:
1.
